conans/client/migrations.py

```diff
@@ -47,7 +47,7 @@ def save_new():
         self.out.warn("Migration: Updating settings.yml")
         if hasattr(migrations_settings, var_name):
             version_default_contents = getattr(migrations_settings, var_name)
-            if version_default_contents != get_default_settings_yml():
+            if version_default_contents.splitlines() != get_default_settings_yml().splitlines():
```

Comment:
How did the line endings became different now?
Shouldn't the lineendings be the same, and not requiring the splitlines?

Reply:
Looks like the __init__.py yaml does not include a new line in the end of the file [after removing some jinja templates](https://github.com/conan-io/conan/pull/8399) for conan 1.34. So now if someone wants to update from any version > 1.33 they will have a settings.yml that is going to be considered different unless we modify the migrations_settings for all versions from 1.33
I thought this may be more reliable for things like this happening in the future and may allow users from 1.33 upgrade, but the other option is also possible.

Reply:
I have added a literal comparison in the test after adding a newline to the generated settings.yml but left the splitlines for checking in the migration if it can be upgraded, wdyt?

Reply:
Good for me 👍